engine-standalone-storage/src/sync/mod.rs

```diff
@@ -424,97 +457,116 @@ fn non_submit_execute<I: IO + Copy>(
 
             Some(TransactionExecutionResult::Submit(Ok(result)))
         }
-
         TransactionKind::DeployErc20(_) => {
             // No promises can be created by `deploy_erc20_token`
             let mut handler = crate::promise::NoScheduler { promise_data };
             let result = contract_methods::connector::deploy_erc20_token(io, env, &mut handler)?;
 
             Some(TransactionExecutionResult::DeployErc20(result))
         }
-
         TransactionKind::FtOnTransfer(_) => {
             // No promises can be created by `ft_on_transfer`
             let mut handler = crate::promise::NoScheduler { promise_data };
             contract_methods::connector::ft_on_transfer(io, env, &mut handler)?;
 
             None
         }
-
         TransactionKind::FtTransferCall(_) => {
-            let mut handler = crate::promise::NoScheduler { promise_data };
-            let promise_args =
-                contract_methods::connector::ft_transfer_call(io, env, &mut handler)?;
+            #[cfg(feature = "ext-connector")]
+            return Ok(None);
 
-            Some(TransactionExecutionResult::Promise(promise_args))
-        }
+            #[cfg(not(feature = "ext-connector"))]
+            {
+                let mut handler = crate::promise::NoScheduler { promise_data };
+                let maybe_promise_args =
+                    contract_methods::connector::ft_transfer_call(io, env, &mut handler)?;
 
+                maybe_promise_args.map(TransactionExecutionResult::Promise)
+            }
+        }
         TransactionKind::ResolveTransfer(_, _) => {
-            let handler = crate::promise::NoScheduler { promise_data };
-            contract_methods::connector::ft_resolve_transfer(io, env, &handler)?;
+            #[cfg(not(feature = "ext-connector"))]
+            {
+                let handler = crate::promise::NoScheduler { promise_data };
+                contract_methods::connector::ft_resolve_transfer(io, env, &handler)?;
+            }
 
             None
         }
-
         TransactionKind::FtTransfer(_) => {
+            #[cfg(not(feature = "ext-connector"))]
```

Comment:
Just wondering, why we don't panic if this kind of transaction isn't supported in this mode?

Reply:
It allows us not to modify the code of the `borealis-engine` (we will do it when split the eth-connector from the main smart contract). It will just skip the eth-connector's transactions if we compile the borealis-engine with the feature `ext-connector`.